onnxruntime/core/providers/cuda/cuda_utils.cu

```diff
@@ -30,13 +30,13 @@ class ConstantBufferImpl : public IConstantBuffer<T> {
 };
 
 template <typename T>
-std::unique_ptr<IConstantBuffer<T>> CreateConstantOnes() {
-  return std::make_unique<ConstantBufferImpl<T>>((T)1);
```

Comment:
>(T)1 [](start = 49, length = 4)

Seems just changing here would fix it?

Reply:
Yes, but need special handling code for half type, need to use MLFloat16 (not T) in ToCudaType for half type. cuda::ToCudaType<MLFloat16>::FromFloat(1.0f); 

---
In reply to: [251551902](https://github.com/Microsoft/onnxruntime/pull/390#discussion_r251551902) [](ancestors = 251551902)

Reply:
You can specialize for half with that conversion.

---
In reply to: [251565369](https://github.com/Microsoft/onnxruntime/pull/390#discussion_r251565369) [](ancestors = 251565369,251551902)